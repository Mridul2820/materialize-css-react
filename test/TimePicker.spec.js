import React from 'react';
import { render } from '@testing-library/react';
import TimePicker from '../src/TimePicker';
import mocker from './helper/new-mocker';

describe('<TimePicker />', () => {
  test('renders', () => {
    const { container } = render(<TimePicker />);
    expect(container).toMatchSnapshot();
  });

  test('renders with a label', () => {
    const { queryByText } = render(<TimePicker label="TimePicker label" />);
    expect(queryByText('TimePicker label')).toBeTruthy();
  });

  describe('js init', () => {
    const timePickerInitMock = jest.fn();
    const timePickerInstanceDestroyMock = jest.fn();
    const timePickerMock = {
      init: (_, options) => {
        timePickerInitMock(options);
        return {
          destroy: timePickerInstanceDestroyMock
        };
      }
    };

    const restore = mocker('Timepicker', timePickerMock);

    beforeEach(() => {
      timePickerInitMock.mockClear();
      timePickerInstanceDestroyMock.mockClear();
    });

    afterAll(() => {
      restore();
    });

    test('uses default options if none are given', () => {
      render(<TimePicker />);

      expect(timePickerInitMock).toHaveBeenCalledWith({
        duration: 350,
        container: null,
        showClearBtn: false,
        defaultTime: 'now',
        fromNow: 0,
        i18n: {
          cancel: 'Cancel',
          clear: 'Clear',
          done: 'Ok'
        },
        autoClose: false,
        twelveHour: true,
        vibrate: true,
        onOpenStart: null,
        onOpenEnd: null,
        onCloseStart: null,
        onCloseEnd: null,
        onSelect: null
      });
    });
  });
});
