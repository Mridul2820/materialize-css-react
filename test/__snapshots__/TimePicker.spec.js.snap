// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimePicker /> renders 1`] = `
<div>
  <div
    class="input-field col"
  >
    <input
      class="timepicker"
      id="TimePicker-mockId"
      type="text"
      value=""
    />
  </div>
</div>
`;
